pkg/config/linters_config.go

```diff
@@ -53,3 +56,11 @@ type RequiredFieldsConfig struct {
 	// When otherwise not specified, the default value is "SuggestFix".
 	PointerPolicy RequiredFieldPointerPolicy `json:"pointerPolicy"`
 }
+
+// StatusSubresourceConfig contains configuration for the statussubresource linter.
+type StatusSubresourceConfig struct {
+	// statusFieldName is the name of the field to check for when
+	// evaluating the presence of a status field.
+	// If not set, the default value is "Status".
+	StatusFieldName string `json:"statusFieldName"`
```

Comment:
Can you set a status subresource, to be a field other than `Status`? 👀

Reply:
IIUC, the json tag should always be `status` but the actual field name in the Go type could, in theory, be anything.

Reply:
Maybe that goes against all of our other assumptions in other linters though, like `jsontags` such that we can reasonably expect that the only name ever valid for the field is literally `Status`

Reply:
The status subresource is looking for a serialised `status` structure within the JSON, in theory the Go field name need not match that. Typically it does, but it's not 100% required and in theory could be anything

Reply:
Updated this to no longer have a configuration option and search for the json tag with a name of `status` to determine if there is a status field in the struct